packages/scene-composer/src/hooks/useRuleResult.ts

```diff
@@ -1,20 +1,34 @@
 import { useMemo } from 'react';
+import { Primitive } from '@iot-app-kit/core';
 
 import { IValueDataBinding } from '../interfaces';
 import { useSceneDocument, useDataStore } from '../store';
 import { useSceneComposerId } from '../common/sceneComposerIdContext';
 import { dataBindingValuesProvider, ruleEvaluator } from '../utils/dataBindingUtils';
 
-const useRuleResult = (ruleMapId: string, dataBinding: IValueDataBinding) => {
+import useBindingData from './useBindingData';
+
+const useRuleResult = ({
+  ruleMapId,
+  dataBinding,
+  defaultValue,
+}: {
+  ruleMapId?: string;
```

Comment:
you can make them as type safe by having them in an interface.

Reply:
it should be the same to define type inline VS named interface?

Reply:
It is more readable and can be reused if you have an interface.